subprojects/tooling-api/src/main/java/org/gradle/tooling/model/build/BuildEnvironment.java

```diff
@@ -65,4 +67,12 @@
      * @throws UnsupportedMethodException For Gradle versions older than 1.0-milestone-8, where this method is not supported.
      */
     JavaEnvironment getJava() throws UnsupportedMethodException;
+
+    /**
+     * Returns an unmodifiable map of the current system environment variables.
+     *
+     * @return the environment as a map of variable names to values
+     * @since 3.5
+     */
+    Map<String, String> getEnvironmentVariables();
```

Comment:
why do we need this? wouldn't this just be a copy of the current process' environment variables, ie those available via `System.getenv()`?